pkg/util/test/util.go

```diff
@@ -2002,6 +2002,49 @@ func ValidateTelemetryInstalled(pxImageList map[string]string, cluster *corev1.S
 	return nil
 }
 
+// ValidatePodTopologySpreadConstraints validates pod topology spread constraints for following component deployments:
+// csi, pvc controller, stork, stork controller
+func ValidatePodTopologySpreadConstraints(t *testing.T, namespace string, topologyKeys []string, timeout, interval time.Duration) {
+	topologyKeySet := make(map[string]bool)
+	for _, key := range topologyKeys {
+		topologyKeySet[key] = true
+	}
+
+	validateDeploymentTopology := func(deploymentName string) func() (interface{}, bool, error) {
```

Comment:
Instead of iterating over all the different deployments, we should call this validate function from individual component's Validate function. So this gets executed every time we call ValidateStorageCluster (or the ValidateComponents, ValidateStork, ValidateCSI, etc). On clouds it will be checked for other tests as well automatically.

We can always generate the list of topologyKeys on the fly looking at the current nodes, instead of the test supplying it.